app/src/main/java/org/vss/impl/postgres/PostgresBackendImpl.java

```diff
@@ -0,0 +1,131 @@
+package org.vss.impl.postgres;
+
+import com.google.inject.Inject;
+import com.google.protobuf.ByteString;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.inject.Singleton;
+import org.jooq.DSLContext;
+import org.jooq.Insert;
+import org.jooq.Query;
+import org.jooq.Update;
+import org.vss.GetObjectRequest;
+import org.vss.GetObjectResponse;
+import org.vss.KVStore;
+import org.vss.KeyValue;
+import org.vss.ListKeyVersionsRequest;
+import org.vss.ListKeyVersionsResponse;
+import org.vss.PutObjectRequest;
+import org.vss.PutObjectResponse;
+import org.vss.exception.ConflictException;
+import org.vss.postgres.tables.records.VssDbRecord;
+
+import static org.jooq.impl.DSL.val;
+import static org.vss.postgres.tables.VssDb.VSS_DB;
+
+@Singleton
+public class PostgresBackendImpl implements KVStore {
+
+  private final DSLContext context;
+
+  @Inject
+  public PostgresBackendImpl(DSLContext context) {
+    this.context = context;
+  }
+
+  @Override
+  public GetObjectResponse get(GetObjectRequest request) {
+
+    VssDbRecord vssDbRecord = context.selectFrom(VSS_DB)
+        .where(VSS_DB.STORE_ID.eq(request.getStoreId())
+            .and(VSS_DB.KEY.eq(request.getKey())))
+        .fetchOne();
+
+    if (vssDbRecord != null) {
+      return GetObjectResponse.newBuilder()
+          .setValue(KeyValue.newBuilder()
+              .setKey(vssDbRecord.getKey())
+              .setValue(ByteString.copyFrom(vssDbRecord.getValue()))
+              .setVersion(vssDbRecord.getVersion())
+              .build())
+          .build();
+    } else {
+      return GetObjectResponse.newBuilder()
+          .setValue(KeyValue.newBuilder()
+              .setKey(request.getKey()).build())
+          .build();
+    }
+  }
+
+  @Override
+  public PutObjectResponse put(PutObjectRequest request) {
+
+    String storeId = request.getStoreId();
+
+    List<VssDbRecord> vssRecords = new ArrayList<>(request.getTransactionItemsList().stream()
+        .map(kv -> buildVssRecord(storeId, kv)).toList());
+
+    VssDbRecord globalVersionRecord = buildVssRecord(storeId,
+        KeyValue.newBuilder()
+            .setKey(GLOBAL_VERSION_KEY)
+            .setVersion(request.getGlobalVersion())
```

Comment:
this sets it unconditionally to the user's input and does not handle `null`.  but I thought the logic should be:

- if not specified (i.e. null), increment the current value and don't check it
- if specified, check the current value, abort if different, otherwise increment it

Reply:
Yes it should be 
* if not specified (i.e. null), do nothing with it, ignore
* if specified, check the current value, abort if different, otherwise increment it